src/Network/Network/NatGateway/NewAzureRMNatGatewayCommand.cs

```diff
@@ -0,0 +1,181 @@
+﻿// <auto-generated>
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// 
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// 
+// 
+// Warning: This code was generated by a tool.
+// 
+// Changes to this file may cause incorrect behavior and will be lost if the
+// code is regenerated.
+// 
+// For documentation on code generator please visit
+//   https://aka.ms/nrp-code-generation
+// Please contact wanrpdev@microsoft.com if you need to make changes to this file.
+// </auto-generated>
+
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using System.Net;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using AutoMapper;
+using MNM = Microsoft.Azure.Management.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NatGateway", SupportsShouldProcess = true), OutputType(typeof(PSNatGateway))]
+    public partial class NewAzureRmNatGateway : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource group name of the nat gateway.",
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("ResourceName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the nat gateway.",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The idle timeout of the nat gateway.",
+            ValueFromPipelineByPropertyName = true)]
+        public int IdleTimeoutInMinutes { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of a NAT gateway SKU.")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter(MNM.NatGatewaySkuName.Standard)]
```

Comment:
AFAIK server sets it to `Basic` currently if not specified. And Basic is not supported yet.
please consider temporary defaulting to `Standard` in cmdlet code or other applicable workaround